File: testdata/jennies/rawtypes/struct_with_defaults/PythonRawTypes/models/defaults.py
import typing


class SomeStruct:
    field_bool: bool
    field_string: str
    field_string_with_constant_value: typing.Literal["auto"]
    field_float32: float
    field_int32: int

    def __init__(self, field_bool: bool = True, field_string: str = "foo", field_float32: float = 42.42, field_int32: int = 42):
        self.field_bool = field_bool
        self.field_string = field_string
        self.field_string_with_constant_value = "auto"
        self.field_float32 = field_float32
        self.field_int32 = field_int32

    def to_json(self) -> dict[str, object]:
        payload: dict[str, object] = {
            "fieldBool": self.field_bool,
            "fieldString": self.field_string,
            "FieldStringWithConstantValue": self.field_string_with_constant_value,
            "FieldFloat32": self.field_float32,
            "FieldInt32": self.field_int32,
        }
        return payload

    @classmethod
    def from_json(cls, data: dict[str, typing.Any]) -> typing.Self:
        args: dict[str, typing.Any] = {}
        
        if "fieldBool" in data:
            args["field_bool"] = data["fieldBool"]
        if "fieldString" in data:
            args["field_string"] = data["fieldString"]
        if "FieldFloat32" in data:
            args["field_float32"] = data["FieldFloat32"]
        if "FieldInt32" in data:
            args["field_int32"] = data["FieldInt32"]        

        return cls(**args)

File: testdata/jennies/rawtypes/struct_with_defaults/RustRawTypes/src/types/defaults.rs
use serde::{Deserialize, Serialize};

#[derive(Serialize, Deserialize, Debug, Clone, PartialEq)]
pub struct SomeStruct {
    #[serde(rename = "fieldBool")]
    pub field_bool: bool,

    #[serde(rename = "fieldString")]
    pub field_string: String,

    #[serde(rename = "FieldStringWithConstantValue")]
    #[serde(default = "default_some_struct_field_string_with_constant_value")]
    pub field_string_with_constant_value: String,

    #[serde(rename = "FieldFloat32")]
    pub field_float32: f32,

    #[serde(rename = "FieldInt32")]
    pub field_int32: i32,
}

impl Default for SomeStruct {
    fn default() -> Self {
        Self {
            field_bool: true,
            field_string: "foo".to_string(),
            field_string_with_constant_value: "auto".to_string(),
            field_float32: 42.42,
            field_int32: 42,
        }
    }
}

fn default_some_struct_field_string_with_constant_value() -> String {
    "auto".to_string()
}
